docs/incubating/custom-evaluator.md

```diff
@@ -0,0 +1,70 @@
+# Custom violations evaluator (incubating)
+
+> Since: [`0.5.1`](https://github.com/novoda/gradle-static-analysis-plugin/releases/tag/v0.5.1)
+>
+> :warning: This is an **experimental**, incubating feature that may be subject to breaking API changes at any time!
+
+The plugin uses a [`ViolationsEvaluator`][violationsevaluatorcode] to determine what to do with the results collected from all the active
+tools (if any). The built-in behaviour is provided by the [`DefaultViolationsEvaluator`][defaultviolationsevaluatorcode], which you can
+read more about [below](#the-defaultviolationsevaluator). The plugin's violations evaluation behaviour is not fixed, and it can be
+customised by providing an implementation of the [`ViolationsEvaluator`][violationsevaluatorcode] interface.
+
+## Table of contents
+ * [The `DefaultViolationsEvaluator`](#the-defaultviolationsevaluator)
+ * [Creating a custom violations evaluator](#creating-a-custom-violations-evaluator)
+
+---
+
+## The `DefaultViolationsEvaluator`
+The plugin has a default mechanism to decide whether to consider a build as passing or failed. The mechanism is manifesting itself
+as the `penalties` closure:
+
+```gradle
+staticAnalysis {
+    penalties {
+        maxErrors = 0
+        maxWarnings = 10
+    }
+    //...
+}
+```
+
+This closure instructs the plugin to use a [`DefaultViolationsEvaluator`][defaultviolationsevaluatorcode] that will count the number of
+errors and warnings and compare them against the set thresholds. For more details, see the
+[Configurable failure thresholds](../advanced-usage.md#configurable-failure-thresholds) documentation.
+
+## Creating a custom violations evaluator
+In order to provide a custom evaluator, you can implement the [`ViolationsEvaluator`][violationsevaluatorcode] interface and provide
+that implementation to the `evaluator` property of the `staticAnalysis` closure. The [`ViolationsEvaluator`][violationsevaluatorcode]
+can be provided as a closure as well:
+
+```gradle
+staticAnalysis {
+	evaluator {	Set<Violations> allViolations ->
```

Comment:
This piece of code uses tabs 😄 Also before `Set` we have tabs, maybe we should have a single space there.

Reply:
I copy-pasted it from Antonio's PR! I swear :D But nice catch, will fix

Reply:
@mr-archano prefers tabs over spaces. 

~[](https://media.giphy.com/media/4ORDH8n5cdPA4/giphy.gif)

Reply:
17d17d2d010d77b9215fb4345dc52f7dfc6079c8